skrub/_agg_joiner.py

```diff
@@ -0,0 +1,592 @@
+        self.tables = tables
+        self.main_key = main_key
+        self.agg_ops = agg_ops
+        self.suffixes = suffixes
+
+    def fit(self, X, y=None) -> "AggJoiner":
+        """Aggregate auxiliary tables based on the main keys.
+
+        Parameters
+        ----------
+        X : {pandas.Dataframe}
+            Input data, based table on which to left join the
+            auxiliary tables.
+        y : array-like of shape (n_samples), default=None
+            Used to compute the correlation between the generated covariates
+            and the target for screening purposes.
+
+        Returns
+        -------
+        AggJoiner
+            Fitted :class:`AggJoiner` instance (self).
+        """
+        self.check_input(X)
+        agg_px, _ = get_namespace([table for (table, _, _) in self.tables_])
+
+        if self.agg_ops is None:
+            agg_ops = ["mean", "mode"]
+        else:
+            agg_ops = np.atleast_1d(self.agg_ops).tolist()
+        self.agg_ops_ = agg_ops
+
+        num_ops, categ_ops = split_num_categ_ops(agg_ops)
+
+        self.agg_tables_ = []
+        for (table, cols_to_join, cols_to_agg), suffix in zip(
+            self.tables_, self.suffixes_
+        ):
+            agg_table = agg_px.aggregate(
+                table,
+                cols_to_join,
+                cols_to_agg,
+                num_ops,
+                categ_ops,
+                suffix,
+            )
+            agg_table = self._screen(agg_table, y)
+            self.agg_tables_.append((agg_table, cols_to_join))
+
+        return self
+
+    def transform(self, X):
+        """Left-join pre-aggregated tables on `X`.
+
+        Parameters
+        ----------
+        X : dataframe,
+            The input data to transform.
+
+        Returns
+        -------
+        X_transformed : dataframe,
+            The augmented input.
+        """
+
+        check_is_fitted(self, "agg_tables_")
+        agg_px, _ = get_namespace([table for (table, _, _) in self.tables_])
+
+        for main_key, (aux, aux_key) in zip(self.main_keys_, self.agg_tables_):
+            X = agg_px.join(
+                left=X,
+                right=aux,
+                left_on=main_key,
+                right_on=aux_key,
+            )
+
+        return X
+
+    def _screen(self, agg_table, y):
+        """Only keep aggregated features which correlation with
+        y is above some threshold.
+        """
+        # TODO: Add logic
+        return agg_table
+
+    def check_input(self, X):
+        """Perform a check on column names data type and suffixes.
+
+        Parameters
+        ----------
+        X : dataframe
+            The raw input to check.
+        """
+        if not hasattr(X, "__dataframe__"):
+            raise TypeError(f"X must be a dataframe, got {type(X)}.")
+
+        # Check main_keys
+        # XXX: Do we want to enable main_keys with multiple columns?
+        # e.g. [["user_id", "movie_id"], ["user_id"]]
```

Comment:
```suggestion
```
As we are not supporting this in this PR or in the immediate future, let's remove the XXX. We can always come back to it later.